File: scripts/eoolo.c
#include <stdio.h>

int expr1();
int expr2();

int main (int argc, char *args[])
{
    printf("%d\n", (expr1() || expr2()));
    return 0;
}

int expr1()
{
    printf("method expr1\n");
    return 1;  // true
}

int expr2()
{
    printf("method expr2\n");
    return 1;  // true
}

File: scripts/eoolo.dart
#!/usr/bin/dart

void main() {
    // In Dart, only the boolean value true is true.
    // All other values are treated as false.
    // So, we use the and operator.
    print(expr1() && expr2());
}

String expr1() {
    print('method expr1');
    return 'return expr1';
}

String expr2() {
    print('method expr2');
    return 'return expr2';
}
